internal/controller/kustomization_controller.go

```diff
@@ -956,10 +956,17 @@ func (r *KustomizationReconciler) prune(ctx context.Context,
 	return false, nil
 }
 
+func finalizerShouldDeleteDependents(obj *kustomizev1.Kustomization) bool {
+	if obj.Spec.DeletionPolicy == nil || *obj.Spec.DeletionPolicy == kustomizev1.DeletionPolicyMirrorPrune {
+		return obj.Spec.Prune
+	}
```

Comment:
Please add a function to the API package called `GetDeletionPolicy()`, similar to `GetRetryInterval()`, that returns the default value when the field value is set to string empty. Then change the logic here to `if obj.GetDeletionPolicy() == kustomizev1.DeletionPolicyMirrorPrune`.

Reply:
This should be fixed now. I will write docs and add more tests before rebasing and squashing everything into a single commit.